### PR TITLE
Add debug option to pool

### PR DESCRIPTION
**Pull request by JedMeister**

### CODE REVIEW

File: pool_bin [hunk 1]
```diff
@@ -582,6 +589,9 @@ def main():
     if args.cmd:
         func = args.func
         args = vars(args)
+        # only pass debug for pool-get
+        if func.__name__ != 'get' and "debug" in args:
+            del args["debug"]
```
Comment (OnGle) [inline pool_bin:594]: Should probably warn that unused args are unused imo

File: pool_lib/__init__.py [hunk 2]
```diff
@@ -262,27 +272,31 @@ def __init__(self, path: AnyPath):
...
         super().__init__(path)
         if recursed_paths is None:
             recursed_paths = []
 
         if self.link in recursed_paths:
             raise CircularDependency(
-                "circular dependency detected `%s' is in recursed paths %s"
-                % (self.link, recursed_paths)
+                f"circular dependency detected `{self.link}' is in recursed"
+                " paths recursed_paths"
```
Comment (OnGle) [inline pool_lib/__init__.py:299]: Old version printed recursed paths, new version doesn't

File: pool_lib/__init__.py [hunk 3]
```diff
@@ -612,7 +640,8 @@ def register(self, stock_ref: str) -> None:
             stock_name += "#" + branch
 
         if stock_name in self.stocks:
-            raise PoolError("stock already registered under name `%s'" % stock_name)
+            raise PoolError(f"stock already registered under name"
```
Comment (OnGle) [inline pool_lib/__init__.py:643]: fstring with no formatting

File: pool_lib/__init__.py [hunk 4]
```diff
@@ -896,7 +932,8 @@ def resolve(self, unresolved: RS) -> RS:
             logger.debug(f"resolve {name=} {version=}")
             if not version:
                 if name not in packages:
-                    raise PoolError("can't resolve non-existent package `%s'" % name)
+                    raise PoolError(f"can't resolve non-existent package"
```
Comment (OnGle) [inline pool_lib/__init__.py:935]: fstring with no formatting

**Review by OnGle — CHANGES_REQUESTED** [review]
See notes

File: pool_bin (hunk 1 above)
Comment (JedMeister) [reply pool_bin:594]: Yeah, that's fair. What would you suggest? Just a `print()`? Or should I use the logger?

File: pool_lib/__init__.py (hunk 2 above)
Comment (JedMeister) [reply pool_lib/__init__.py:299]: Great catch, thanks.

File: pool_lib/__init__.py (hunk 3 above)
Comment (JedMeister) [reply pool_lib/__init__.py:643]: Another good catch! :)

File: pool_bin (hunk 1 above)
Comment (JedMeister) [reply pool_bin:594]: Just had a closer look (after our offline chat about this) and it appears that the `--debug` switch only applies to `pool-get` already. If you try to use `--debug` with other commands, you get an error. E.g.:
```
pool-info --debug
usage: pool [-h] {exists,gc,get,info-get,info,init,list,register,unregister} ...
pool: error: unrecognized arguments: --debug
```

**Pull request merged**